projects/Era-file-support-for-lodestar.md

```diff
@@ -0,0 +1,82 @@
+# Adding Era File Support in Lodestar
+
+Implement support for `.era` files in Lodestar to enable efficient long-term storage and distribution of Ethereum consensus data.
+
+## Motivation
+
+Lodestar currently supports consensus layer operations but does not yet natively integrate with the `e2store` / `era` archival format. Era files are specialized `.e2s` files that bundle states and blocks in fixed-size groups (one per `SLOTS_PER_HISTORICAL_ROOT`). They provide a canonical cold-storage representation of Ethereum history, optimized for reading, validation, and checkpoint sync.
+
+Native Lodestar support for `.era` files unlocks several benefits:
+- **Historical sync efficiency**: Ability to bootstrap from finalized checkpoints using pre-verified era files instead of downloading full block/state histories.
+- **Validator friendliness**: Era files allow lightweight devices to access verified history without recomputation.
+- **Interoperability**: Ensures Lodestar can consume and produce files compatible with other clients like Nimbus, enabling ecosystem-wide standardization.
+- **Auditability and distribution**: Researchers, light clients, and custodians can rely on `.era` files for fast consistency checks.
+
+## Project description
+
+The project involves implementing reading, verification, and  writing of `.era` files in Lodestar.
+
+### Scope
+
+- **Parsing `.e2s` records**: Implement utilities to read the 8-byte headers .
+- **Era grouping**: Add logic to parse `.era` file groups, which include a set of blocks, an era-ending state, and trailing indices.
+- **Verification**:
+  - Ensure block roots match those recorded in the state.
+  - Verify consistency of indices and offsets.
+  - Confirm historical root correctness and alignment with checkpoints.
+- **Integration with Lodestar sync**:
+  - Allow importing an era file as a sync source.
+  - Enable checkpoint sync from an era state root.
+  - Provide APIs to iterate over blocks and states in an era file.
+
+### Deliverables
+
+- A Lodestar module (`packages/era`) for `.e2s`/`.era` parsing and verification.
+- Utilities for inspecting and validating era files from CLI.
+- Hooks in Lodestar sync to use `.era` files as data sources.
+
+## Work during the fellowship
```

Comment:
This part is missing specification. Some more details on how do you implement features mentioned above

Reply:
You can elaborate on which part of the client this fits in, what functions, libs etc